golang/testsuite/networks_impl/test_network.go

```diff
@@ -19,54 +19,53 @@ import (
 
 const (
 	datastoreServiceId services.ServiceID = "datastore"
-	apiServiceIdPrefix = "api-"
+	apiServiceIdPrefix                    = "api-"
 
 	waitForStartupDelayMilliseconds = 1000
-	waitForStartupMaxNumPolls = 15
+	waitForStartupMaxNumPolls       = 15
 )
 
 //  A custom Network implementation is intended to make test-writing easier by wrapping low-level
 //    NetworkContext calls with custom higher-level business logic
 type TestNetwork struct {
-	networkCtx            *networks.NetworkContext
-	datastoreServiceImage string
-	apiServiceImage       string
-	datastoreService      *datastore.DatastoreService
-	personModifyingApiService *api.ApiService
-	personRetrievingApiService *api.ApiService
-	nextApiServiceId      int
+	networkCtx                *networks.NetworkContext
+	datastoreServiceImage     string
+	apiServiceImage           string
+	datastoreIPAddress        string
+	datastorePort             int
+	personModifyingApiClient  *api_service_client.APIClient
+	personRetrievingApiClient *api_service_client.APIClient
+	nextApiServiceId          int
 }
 
 func NewTestNetwork(networkCtx *networks.NetworkContext, datastoreServiceImage string, apiServiceImage string) *TestNetwork {
 	return &TestNetwork{
-		networkCtx:            networkCtx,
-		datastoreServiceImage: datastoreServiceImage,
-		apiServiceImage:       apiServiceImage,
-		datastoreService:      nil,
-		personModifyingApiService: nil,
-		personRetrievingApiService: nil,
-		nextApiServiceId:      0,
+		networkCtx:                networkCtx,
+		datastoreServiceImage:     datastoreServiceImage,
+		apiServiceImage:           apiServiceImage,
+		datastoreIPAddress:        "",
```

Comment:
j/w, why are we replacing `datastoreService` with `datastoreIp/Port` rather than `datastoreClient`?

Reply:
Because `datastoreClient` has encapsulated those values and doesn't have getters. So, I thought that it was the same as putting the values, but if you prefer I can modify `example microservices` and put getters in datastoreClient. 
Let me know what do you prefer?

Reply:
Yep definitely think getters is the way to go here - I generally follow the rule that "any data which is provided in the constructor of the class should a) be generally immutable and b) be accessible via getters if required" (doesn't mean you HAVE to provide getters, and I prefer not to by default - in this case we need them, so providing them makes sense to me)